core/src/test/java/org/elasticsearch/action/bulk/TransportBulkActionIngestTests.java

```diff
@@ -200,6 +235,38 @@ public void testIngestLocal() throws Exception {
         verifyZeroInteractions(transportService);
     }
 
+    public void testSingleItemBulkActionIngestLocal() throws Exception {
+        Exception exception = new Exception("fake exception");
+        IndexRequest indexRequest = new IndexRequest("index", "type", "id");
+        indexRequest.source(Collections.emptyMap());
+        indexRequest.setPipeline("testpipeline");
+        AtomicBoolean responseCalled = new AtomicBoolean(false);
+        AtomicBoolean failureCalled = new AtomicBoolean(false);
+        singleItemBulkWriteAction.execute(null, indexRequest, ActionListener.wrap(
+                response -> {
+                    responseCalled.set(true);
+                },
+                e -> {
+                    assertThat(e, sameInstance(exception));
+                    failureCalled.set(true);
+                }));
+
+        // check failure works, and passes through to the listener
+        assertFalse(action.isExecuted); // haven't executed yet
+        assertFalse(responseCalled.get());
+        assertFalse(failureCalled.get());
+        verify(executionService).executeBulkRequest(bulkDocsItr.capture(), failureHandler.capture(), completionHandler.capture());
+        completionHandler.getValue().accept(exception);
+        assertTrue(failureCalled.get());
+
+        // now check success
+        indexRequest.setPipeline(null); // this is done by the real pipeline execution service when processing
+        completionHandler.getValue().accept(null);
+        assertTrue(action.isExecuted);
+        assertFalse(responseCalled.get()); // listener would only be called by real index action, not our mocked one
+        verifyZeroInteractions(transportService);
```

Comment:
I think we should also verify that interaction happened with `ingestService` mock here?